Dockerfile.onbuild

```diff
@@ -10,20 +10,18 @@ ONBUILD ARG YARN_VERSION
 # Allow installation as non-root user
 ONBUILD RUN npm config set unsafe-perm true
 
-# All but package.json is optional
-ONBUILD COPY package.json yarn.lock* .yarnrc* .npmrc* npm-shrinkwrap.json* package-lock.json* pnpm-lock.yaml* ./
+ONBUILD COPY . ./
 
 # Install dependencies for native builds
 # This is in one giant command to keep the image size small
 # TODO: When Finnbuild uses Docker 1.13, we can use --squash, which means this won't have to be one giant command
 ONBUILD RUN apk upgrade -U && \
     apk add --no-cache --virtual build-dependencies make gcc g++ python git || \
     apk add --no-cache --virtual build-dependencies make gcc g++ python3 git && \
+    corepack enable && \
+    corepack prepare --activate --all && \
     install-dependencies.sh && \
-    rm /usr/local/bin/yarn && npm uninstall --loglevel warn --global pnpm && npm uninstall --loglevel warn --global npm && \
     apk del build-dependencies
 
-ONBUILD COPY . ./
```

Comment:
Possibly better to do the copy as late as possible so Docker can reuse the layer which installs build dependencies, corepack and node dependencies when only changing some production code (this has nothing to do with package managers).

Reply:
Then I think we need to copy any lock file here and all the other files later. I am not 100% sure about that but we definetly need package.json copied here before we run any of the commands. I would really love if we could make it so that we don't have to hard code each package managers lock files into this. Any suggestions on how to avoid hard coding such?